suave/e2e/workflow_test.go

```diff
@@ -1050,6 +1050,54 @@ func TestRelayBlockSubmissionContract(t *testing.T) {
 	require.True(t, ok)
 }
 
+func TestE2E_EmitLogs_Basic(t *testing.T) {
+	fr := newFramework(t, WithKettleAddress())
+	defer fr.Close()
+
+	clt := fr.NewSDKClient()
+
+	// We reuse the same address for both the source and target contract
+	contractAddr := common.Address{0x3}
+	sourceContract := sdk.GetContract(contractAddr, exampleCallSourceContract.Abi, clt)
+
+	res, err := sourceContract.SendTransaction("emitLog", []interface{}{}, nil)
+	require.NoError(t, err)
+
+	fr.suethSrv.ProgressChain()
+
+	receipt, err := res.Wait()
+	require.NoError(t, err)
+	require.Equal(t, receipt.Status, uint64(1))
+
+	require.Len(t, receipt.Logs, 1)
+	require.Equal(t, receipt.Logs[0].Address, contractAddr)
+
+	data, err := exampleCallSourceContract.Abi.Events["LogCallback"].Inputs.Unpack(receipt.Logs[0].Data)
+	require.NoError(t, err)
+
+	execResultBytes := data[0].([]byte)
+
+	var execResult suave.ExecResult
+	require.NoError(t, execResult.DecodeABI(execResultBytes))
+
+	require.Len(t, execResult.Logs, 5)
+
+	t.Log("Testcases for execution result encoding")
+	t.Log(hex.EncodeToString(execResultBytes))
+
+	// check topics size for each log
+	require.Len(t, execResult.Logs[0].Topics, 0)
+	require.Len(t, execResult.Logs[1].Topics, 1)
+	require.Len(t, execResult.Logs[2].Topics, 2)
+	require.Len(t, execResult.Logs[3].Topics, 3)
+	require.Len(t, execResult.Logs[4].Topics, 4)
```

Comment:
It'd be great to also see how you can decode the specific log
Ideally we get the actual logs rather than the LogCallback containing data

Reply:
The logs at that point are already decoded. It is done in https://github.com/flashbots/suave-geth/pull/211/files#diff-04dd0655f18243b3d4950eed741c8b50d9868ff52727bc3da5d9f990c46aa061R1081.

Reply:
Oooh